rust/arrow/src/buffer.rs

```diff
@@ -270,6 +270,22 @@ impl Buffer {
         BitChunks::new(&self, offset, len)
     }
 
+    /// Returns the number of 1-bits in `data`
+    pub fn count_set_bits(&self) -> usize {
+        let len_in_bits = self.len() * 8;
+        self.count_set_bits_offset(0, len_in_bits)
```

Comment:
This seems to ignore `self.offset` (it is on purpose, right?) If yes, do you think that it makes sense to add a test verifying that the offset does not impact this, maybe we already test that for `self.bit_chunks`?

Reply:
Yes, `self.offset` is already considered by `bit_chunks`, or rather by the `data` and `raw_data` functions. I added a test and comment about that.